### PR TITLE
feat: add new guild helper methods

### LINKED ISSUE #646: [REQUEST] add special Member fetching Guild helper methods
### Describe the feature.

I would like to stop needing to use _http to get a list of guild members or search for a guild member (didn't actually know that existed before, very interesting function, may be cool for autocompleting), so having guild helper methods would be very nice. Possibly like 

`my_guild.get_list_of_members(limit = 500)` (kind of a mouthful, not sure if it would be good to rename this)
`my_guild.search_guild_members('Dworv')` (also a bit long, not sure what to do about it)

thoughts?

### Code of Conduct

- [x] I agree to follow the contribution requirements.

### PR DESCRIPTION
**Pull request by r-warfield**
## About

This pull request is about two new guild helper methods for some uncommon http methods

New methods:
`Guild.get_list_of_members()`
`Guild.search_members()`

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): #646 
- [x] I've made this pull request for/as: (check all that apply)
  - [x] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Title.

**Comment by r-warfield** [comment]
> Title.

oop ty

**Comment by r-warfield** [comment]
> Title.

Resolved, you can resolve or whatever now

File: interactions/api/models/guild.pyi [hunk 1]
```diff
@@ -394,6 +394,16 @@ class Guild(DictSerializerMixin):
         emoji: Union[Emoji, int],
         reason: Optional[str] = None,
     ) -> None: ...
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[int] = None,
+    ) -> list: ...
```
Comment (sillyElvy) [inline interactions/api/models/guild.pyi]:
```suggestion
    ) -> List[Member]: ...
```

File: interactions/api/models/guild.pyi [hunk 2]
```diff
@@ -394,6 +394,16 @@ class Guild(DictSerializerMixin):
...
+    async def search_members(
+        self,
+        query: str,
+        limit: Optional[int] = 1
+    ) -> list: ...
```
Comment (sillyElvy) [inline interactions/api/models/guild.pyi]:
```suggestion
    ) -> List[Member]: ...
```

File: interactions/api/models/guild.py [hunk 3]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
...
+        :type after: Optional[int]
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        res = await self._client.get_list_of_members(
+            guild_id=int(self.id), limit=limit, after=after
+        )
+        return [Member(**member, _client=self._client) for member in res]
+
+    async def search_members(self, query: str, limit: Optional[int] = 1) -> list:
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
    async def search_members(self, query: str, limit: Optional[int] = 1) -> List[Member]:
```

File: interactions/api/models/guild.py [hunk 4]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
             reason=reason,
         )
 
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[int] = None,
+    ) -> list:
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
    ) -> List[Member]:
```

File: interactions/api/models/guild.py [hunk 5]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
...
+        """
+        Lists the members of a guild.
+
+        :param limit?: How many members to get from the API. Max is 1000.
+        :type limit: Optional[int]
+        :param after?: Get Member IDs after this snowflake.
+        :type after: Optional[int]
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]: Add `:return:` and `:rtype:`

File: interactions/api/models/guild.py [hunk 6]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
...
+
+    async def search_members(self, query: str, limit: Optional[int] = 1) -> list:
+        """
+        Search the guild for members whose username or nickname starts with provided string.
+
+        :param query: The string to search for
+        :type query: str
+        :param limit?: The number of members to return.
+        :type limit: Optional[int]
+        """
```
Comment (sillyElvy) [inline interactions/api/models/guild.py:1585]: Add `:return:` and `:rtype:`

File: interactions/api/models/guild.py [hunk 7]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
...
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[int] = None,
+    ) -> list:
+        """
+        Lists the members of a guild.
+
+        :param limit?: How many members to get from the API. Max is 1000.
+        :type limit: Optional[int]
+        :param after?: Get Member IDs after this snowflake.
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        :param after?: Get only Members after this member.
```

File: interactions/api/models/guild.py [hunk 8]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
             reason=reason,
         )
 
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[int] = None,
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        after: Optional[Union[Member, int] = MISSING,
```

File: interactions/api/models/guild.py (hunk 5 above)
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        :type after: Optional[Union[Member, int]
```

File: interactions/api/models/guild.py [hunk 9]
```diff
@@ -1546,6 +1546,42 @@ async def delete_emoji(
...
+
+        :param limit?: How many members to get from the API. Max is 1000.
+        :type limit: Optional[int]
+        :param after?: Get Member IDs after this snowflake.
+        :type after: Optional[int]
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        res = await self._client.get_list_of_members(
+            guild_id=int(self.id), limit=limit, after=after
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        if after is not MISSING:
                _after = int(after.id) if not isinstance(after, int) else after
        else:
                _after = None
        res = await self._client.get_list_of_members(
            guild_id=int(self.id), limit=limit, after=_after
```

File: interactions/api/models/guild.pyi [hunk 10]
```diff
@@ -394,6 +394,16 @@ class Guild(DictSerializerMixin):
         emoji: Union[Emoji, int],
         reason: Optional[str] = None,
     ) -> None: ...
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[int] = None,
```
Comment (sillyElvy) [inline interactions/api/models/guild.pyi]:
```suggestion
        after: Optional[Union[Member, int] = MISSING,
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/guild.py (hunk 5 above)
Comment (r-warfield) [reply interactions/api/models/guild.py]: doc: add docstring for returning values in get_list_of_members

Comment (r-warfield) [reply interactions/api/models/guild.py]: bruh

Comment (r-warfield) [reply interactions/api/models/guild.py]: fixed

File: interactions/api/models/guild.py (hunk 6 above)
Comment (r-warfield) [reply interactions/api/models/guild.py:1585]: fixex

File: interactions/api/models/guild.py [hunk 11]
```diff
@@ -1546,6 +1546,50 @@ async def delete_emoji(
             reason=reason,
         )
 
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[Union[Member, int] = MISSING,
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        after: Optional[Union[Member, int]] = MISSING,
```

my bad 😅

File: interactions/api/models/guild.pyi [hunk 12]
```diff
@@ -394,6 +394,16 @@ class Guild(DictSerializerMixin):
         emoji: Union[Emoji, int],
         reason: Optional[str] = None,
     ) -> None: ...
+    async def get_list_of_members(
+        self,
+        limit: Optional[int] = 1,
+        after: Optional[Union[Member, int] = MISSING,
```
Comment (sillyElvy) [inline interactions/api/models/guild.pyi]:
```suggestion
        after: Optional[Union[Member, int]] = MISSING,
```

File: interactions/api/models/guild.py [hunk 13]
```diff
@@ -1546,6 +1546,50 @@ async def delete_emoji(
...
+    ) -> List[Member]:
+        """
+        Lists the members of a guild.
+
+        :param limit?: How many members to get from the API. Max is 1000.
+        :type limit: Optional[int]
+        :param after?: Get only Members after this member.
+        :type after: Optional[Union[Member, int]
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        :type after: Optional[Union[Member, int]]
```

**Review by sillyElvy — APPROVED** [review]
i did it on mobile, thats my excuse

**Review by sillyElvy — CHANGES_REQUESTED** [review]
wrong button before

**Review by sillyElvy — APPROVED** [review]
LGTM

**Pull request merged**